core/unit_test/TestWithoutInitializing.hpp

```diff
@@ -223,8 +223,7 @@ TEST(TEST_CATEGORY, deep_copy_zero_memset) {
 
   // for MI300A with unified memory, ZeroMemset uses a parallel for
   auto success =
-#if defined(KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY) && \
-    defined(KOKKOS_ARCH_AMD_GFX942)
+#if defined(KOKKOS_IMPL_HIP_UNIFIED_MEMORY) && defined(KOKKOS_ARCH_AMD_GFX942)
```

Comment:
The only way `KOKKOS_IMPL_HIP_UNIFIED_MEMORY` is defined is for `KOKKOS_ARCH_AMD_GFX942_APU`, right?
```suggestion
#if defined(KOKKOS_IMPL_HIP_UNIFIED_MEMORY)
```